src/compiler/checker.ts

```diff
@@ -15290,9 +15290,10 @@ namespace ts {
             return suggestion && symbolName(suggestion);
         }
 
-        function getSuggestionForNonexistentSymbol(location: Node, name: __String, meaning: SymbolFlags): string {
-            const result = resolveNameHelper(location, name, meaning, /*nameNotFoundMessage*/ undefined, name, /*isUse*/ false, (symbols, name, meaning) => {
-                // `name` from the callback === the outer `name`
+        function getSuggestionForNonexistentSymbol(location: Node, outerName: __String, meaning: SymbolFlags): string {
+            Debug.assert(outerName !== undefined, "outername should always be defined");

```

Comment:
The description is optional -- in this case you're simply repeating what the code says.

Reply:
Should make the assert less mysterious in logs, etc, though, since Javascript can only print `false`, not `outerName !== undefined`, otherwise.